src/core/announcements/serialization.ts

```diff
@@ -8,12 +11,56 @@ import { Announcement } from "./factories";
  * @returns A string serialization of the announcement
  */
 export const serialize = (announcement: Announcement): string => {
-  const sortedObj = sortObject(announcement as unknown as Record<string, unknown>);
-  let serialization = "";
+  return Object.entries(announcement)
+    .map(serializeValue)
+    .sort(([aKey], [bKey]) => aKey.localeCompare(bKey, "us"))
+    .reduce((serialization, [key, value]) => `${serialization}${key}${value}`, "");
+};
 
-  for (const key in sortedObj) {
-    serialization = `${serialization}${key}${sortedObj[key]}`;
+/**
+ * serializeToHex() takes in just about anything and attempts to coerce the value to DSNP hexadecimal standard
+ *
+ * @param value - DSNP User Id/URI BigNumber BigInt number and even unknown
+ * @throws {@link InvalidHexadecimalSerialization}
+ * Thrown if the coercion failed
+ * @returns A hexadecimal serialized value
+ */
+export const serializeToHex = (
+  value: BigNumber | HexString | DSNPUserId | DSNPUserURI | number | bigint | unknown
+): HexString => {
+  if (typeof value === "number" || typeof value === "bigint") {
+    return prefixAndStrip(value.toString(16));
+  }
+  if (BigNumber.isBigNumber(value)) {
+    return prefixAndStrip(value.toHexString());
+  }
+  if (String(value) === "") {
```

Comment:
FWIW I just checked this bit of code with a value created from the Long library, which is what's used by parquet, and it works.